torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -242,6 +242,8 @@ std::vector<TypePtr> SimpleValue::getItersTypeInfo(
     TypePtr val_type = getValue()->type();
     if (auto list_type = val_type->cast<ListType>()) {
       return {list_type->getElementType()};
+    } else if (val_type->isSubtypeOf(TensorType::get())) {
```

Comment:
nitpick: maybe we could add a short comment explaining why `IterType` of a tensor is still a `TensorType`